js_modules/dagit/packages/core/src/assets/LaunchAssetExecutionButton.tsx

```diff
@@ -317,6 +317,18 @@ export const useMaterializationAction = (preferredJobName?: string) => {
           target={state.target}
           open={true}
           setOpen={() => setState({type: 'none'})}
+          refetch={async () => {
+            const result = await client.query<
+              LaunchAssetLoaderQuery,
+              LaunchAssetLoaderQueryVariables
+            >({
+              query: LAUNCH_ASSET_LOADER_QUERY,
+              variables: {assetKeys: state.assets.map(({assetKey}) => ({path: assetKey.path}))},
+            });
+            const assets = result.data.assetNodes;
+            const next = await stateForLaunchingAssets(client, assets, false, preferredJobName);
+            setState(next);
```

Comment:
Not urgent, but it might be nice to handle the next.state === 'error' scenario here. I wonder if we could pull lines 239-262 out of  `onClick` into a separate helper, and have both that onClick method and this one call that to avoid having two places where we need to make the query.